idaes/apps/flexibility_analysis/decision_rules/linear_dr.py

```diff
@@ -0,0 +1,127 @@
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+#################################################################################
+"""
+This module contains a function for constructing a linear decision rule for 
+the inner problem of the flexibility test.
+"""
+from typing import MutableMapping, Sequence
+import pyomo.environ as pe
+from pyomo.core.base.var import _GeneralVarData, ScalarVar, IndexedVar
+from pyomo.core.base.block import _BlockData
+from pyomo.core.expr.numeric_expr import LinearExpression
+from pyomo.common.config import ConfigValue
+from .dr_config import DRConfig
+from ..check_optimal import assert_optimal_termination
+
+
+class LinearDRConfig(DRConfig):
+    r"""
+    A class for specifying options for constructing linear decision rules
+    for use in the flexibility test problem.
+
+    Attributes
+    ----------
+    solver: Union[Solver, OptSolver]
+        The solver to use for building the linear decision rule (an LP solver).
+    """
+
+    def __init__(
+        self,
+        description=None,
+        doc=None,
+        implicit=False,
+        implicit_domain=None,
+        visibility=0,
+    ):
+        super().__init__(
+            description=description,
+            doc=doc,
+            implicit=implicit,
+            implicit_domain=implicit_domain,
+            visibility=visibility,
+        )
+        self.solver = self.declare(
+            "solver", ConfigValue(default=pe.SolverFactory("ipopt"))
+        )
+
+
+def construct_linear_decision_rule(
+    input_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    output_vals: MutableMapping[_GeneralVarData, Sequence[float]],
+    config: LinearDRConfig,
+) -> _BlockData:
+    """
+    Construct a linear decision rule from the data provided for the inputs
+    and outputs.
+
+    Parameters
+    ----------
+    input_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are inputs to the decision rule
+    output_vals: input_vals: MutableMapping[_GeneralVarData, Sequence[float]]
+        Data for the variables that are outputs to the decision rule
+    config: LinearDRConfig
+        A config object to specify options for the decision rule
+
+    Returns
+    -------
+    res: _BlockData
+        A pyomo model containing the linear decision rule
+    """
+    n_inputs = len(input_vals)
+    n_outputs = len(output_vals)
+
+    res = pe.Block(concrete=True)
+    res.output_set = pe.Set(initialize=list(range(n_outputs)))
+    res.decision_rule = pe.Constraint(res.output_set)
+
+    for out_ndx, (output_var, out_samples) in enumerate(output_vals.items()):
```

Comment:
For later maintainers, some inline docs would be helpful. I can work out what each part is doing, but I don't know why it is doing it.